api4/file.go

```diff
@@ -65,32 +66,62 @@ func uploadFile(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := r.ParseMultipartForm(*c.App.Config().FileSettings.MaxFileSize); err != nil {
+	var resStruct *model.FileUploadResponse
+	var appErr *model.AppError
+
+	if err := r.ParseMultipartForm(*c.App.Config().FileSettings.MaxFileSize); err != nil && err != http.ErrNotMultipart {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
-	}
+	} else if err == http.ErrNotMultipart {
+		defer r.Body.Close()
 
-	m := r.MultipartForm
+		c.RequireChannelId()
+		c.RequireFilename()
```

Comment:
No action needed but just a heads-up that you can do `c.RequireChannelId().RequireFilename()` if you wanted